core/services/workflows/v2/engine.go

```diff
@@ -2,26 +2,33 @@ package v2
 
 import (
 	"context"
+	"fmt"
+	"time"
 
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
 	"github.com/smartcontractkit/chainlink-common/pkg/services"
-	"github.com/smartcontractkit/chainlink-common/pkg/workflows/wasm/host"
-	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/workflows/internal"
 )
 
-type EngineConfig struct {
-	Lggr       logger.Logger
-	WorkflowID string
-	Module     host.ModuleV2
-}
-
 type Engine struct {
 	services.Service
 	srvcEng *services.Engine
 
-	cfg EngineConfig
+	cfg       EngineConfig
+	localNode capabilities.Node
+}
+
+type LifecycleHooks struct {
+	OnInitialized       func(success bool)
```

Comment:
_nit_ is it more helpful to have `func(err error)`?  I guess the payload can evolve

Reply:
good idea, thanks